src/wp-admin/includes/plugin.php

```diff
@@ -142,14 +142,8 @@ function _get_plugin_data_markup_translate( $plugin_file, $plugin_data, $markup
 	// Translate fields.
 	if ( $translate ) {
 		$textdomain = $plugin_data['TextDomain'];
-		if ( $textdomain ) {
```

Comment:
Alternative in pseudo:

`$textdomain && ( ! registry->exists( $textdomain ) ) || ( registry->exists( $textdomain ) && registry->has( $textdomain ) ) ) && ! is_textdomain_loaded()`